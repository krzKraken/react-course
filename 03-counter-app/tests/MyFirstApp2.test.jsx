/** @jest-environment jsdom */

//? screen return the html update in the
import { render, screen } from "@testing-library/react";

const { MyFirstApp } = require("../src/myFirstApp");

describe("Testing <MyFirstApp/>", () => {
  const title = "hola, soy Goku"
  const subTitle = "Hola soy un subtitle"
  test('should match with the snapshot', () => {

    const { container } = render(
      <MyFirstApp title={title} />
    );
    expect(container).toMatchSnapshot();
  })

  test('should match with the snapshot', () => {
    render(<MyFirstApp title={title} />)
    // screen.debug(); // -> console.log(screen)
    //? exist the title? 

    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show the title in a h1 tag', () => {
    render(<MyFirstApp title={title} />)
    // screen.debug(); // -> console.log(screen)
    //? exist the title? 

    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  });

  test('should show the subtitle sent by the props', () => {
    render(
      <MyFirstApp
        title={title}
        subTitle={subTitle}
      />
    );
    screen.debug(); // -> console.log(screen)
    expect(screen.getAllByText(subTitle).length).toBe(2);


  });

});
