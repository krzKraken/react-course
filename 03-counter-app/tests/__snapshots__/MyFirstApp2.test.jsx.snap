// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MyFirstApp/> should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola, soy Goku
  </h1>
  <p>
    default Subtitle
  </p>
  <p>
    default Subtitle
  </p>
  <p>
     
    default UserName
  </p>
</div>
`;
